lib/std/heap.zig

```diff
@@ -8,6 +8,158 @@ const c = std.c;
 const Allocator = std.mem.Allocator;
 const windows = std.os.windows;
 
+const page_size_os_arch_unsupported = @compileError("The Zig standard library is missing page_size definitions for the target architecture/OS");
+const page_size_os_unsupported = @compileError("The Zig standard library is missing page_size definitions for the target OS");
+const page_size_query_unsupported = @compileError("The Zig standard library is missing support for querying page size on this OS");
+
+/// This value defines the largest page size for this architecture/OS combination that the standard library allows. The standard library asserts that the `pageSize()` does not exceed `max_page_size`. To allow larger page sizes, override `max_page_size` as well as `-z max-page-size`.
+pub const max_page_size: usize = switch (builtin.os.tag) {
+    .driverkit, .ios, .macos, .tvos, .visionos, .watchos => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 16 << 10,
+        else => page_size_os_arch_unsupported,
+    },
+    // -- <https://devblogs.microsoft.com/oldnewthing/20210510-00/?p=105200>
+    .windows => switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // SuperH => 4 << 10,
+        .mips, .mipsel, .mips64, .mips64el => 4 << 10,
+        .powerpc, .powerpcle, .powerpc64, .powerpc64le => 4 << 10,
+        // DEC Alpha => 8 << 10,
+        // Itanium => 8 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be => 4 << 10,
+        else => page_size_os_arch_unsupported,
+    },
+    // TODO: freestanding, uefi, freebsd, netbsd, dragonfly, openbsd
```

Comment:
@mikdusan would you be able to chip in for FreeBSD?

Maybe someone knows who we could ping for the others?

Reply:
These are low priority I think. Most BSD architectures are in 🔍: "nobody has really looked into this target so whether or not it works is unknown".